kernel/kernel-api/src/test/scala/io/delta/kernel/internal/snapshot/LogSegmentSuite.scala

```diff
@@ -234,4 +236,136 @@ class LogSegmentSuite extends AnyFunSuite with MockFileSystemClientUtils {
     assert(logSegment.toString === expectedToString)
   }
+
+  test("allFilesWithCompactionsReversed -- 3 - 5 in middle") {
+    val logSegment = new LogSegment(
+      logPath,
+      6,
+      deltaFileStatuses(Seq.range(0, 7)).toList.asJava,
+      compactionFs3To5List,
+      Collections.emptyList(),
+      1)
+    val allFiles = logSegment.allFilesWithCompactionsReversed();
+
+    // expect to get 6, 3-5.compact, 2, 1, 0
+    val expected = List(
+      FileStatus.of(FileNames.deltaFile(logPath, 6), 6, 6 * 10),
+      FileStatus.of(FileNames.logCompactionPath(logPath, 3, 5).toString, 3, 3 * 10),
+      FileStatus.of(FileNames.deltaFile(logPath, 2), 2, 2 * 10),
+      FileStatus.of(FileNames.deltaFile(logPath, 1), 1, 1 * 10),
+      FileStatus.of(FileNames.deltaFile(logPath, 0), 0, 0 * 10)).asJava;
+    assert(allFiles === expected);
+  }
+
+  test("allFilesWithCompactionsReversed -- 3 - 5 at start") {
+    val logSegment = new LogSegment(
+      logPath,
+      7,
+      deltaFileStatuses(Seq.range(3, 8)).toList.asJava,
+      compactionFs3To5List,
+      Collections.emptyList(),
+      1)
+    val allFiles = logSegment.allFilesWithCompactionsReversed();
+
+    // expect to get 7, 6, 3-5.compact
+    val expected = List(
+      FileStatus.of(FileNames.deltaFile(logPath, 7), 7, 7 * 10),
+      FileStatus.of(FileNames.deltaFile(logPath, 6), 6, 6 * 10),
+      FileStatus.of(FileNames.logCompactionPath(logPath, 3, 5).toString, 3, 3 * 10)).asJava;
+    assert(allFiles === expected);
+  }
+
+  test("allFilesWithCompactionsReversed -- 3 - 5 at end") {
+    val logSegment = new LogSegment(
+      logPath,
+      5,
+      deltaFileStatuses(Seq.range(0, 6)).toList.asJava,
+      compactionFs3To5List,
+      Collections.emptyList(),
+      1)
+    val allFiles = logSegment.allFilesWithCompactionsReversed();
+
+    // expect to get 3-5.compact, 2, 1, 0
+    val expected = List(
+      FileStatus.of(FileNames.logCompactionPath(logPath, 3, 5).toString, 3, 3 * 10),
+      FileStatus.of(FileNames.deltaFile(logPath, 2), 2, 2 * 10),
+      FileStatus.of(FileNames.deltaFile(logPath, 1), 1, 1 * 10),
+      FileStatus.of(FileNames.deltaFile(logPath, 0), 0, 0 * 10)).asJava;
+    assert(allFiles === expected);
+  }
+
+  test("allFilesWithCompactionsReversed -- 3 - 5 at second to last") {
+    val logSegment = new LogSegment(
+      logPath,
+      6,
+      deltaFileStatuses(Seq.range(2, 7)).toList.asJava,
+      compactionFs3To5List,
+      Collections.emptyList(),
+      1)
+    val allFiles = logSegment.allFilesWithCompactionsReversed();
+
+    // expect to get 6, 3-5.compact, 2
+    val expected = List(
+      FileStatus.of(FileNames.deltaFile(logPath, 6), 6, 6 * 10),
+      FileStatus.of(FileNames.logCompactionPath(logPath, 3, 5).toString, 3, 3 * 10),
+      FileStatus.of(FileNames.deltaFile(logPath, 2), 2, 2 * 10)).asJava;
+    assert(allFiles === expected);
+  }
+
+  test("allFilesWithCompactionsReversed -- 3 - 5, and 7 - 9") {
```

Comment:
could we test some more cases of having multiple CRC?

See far I see
- no overlap (1 delta gap in between)
- overlap

Can we add
- no overlap (0 delta gap in betwee) and multiple compactions, e.g. `3-5, 6-8, 9-11`
- overlap (range contains another) e.g. `3-15, 5-9`
- one super complex case with a bunch of different overlaps and no overlap cases?

Reply:
These are all added now